### PR TITLE
[hadoop][cluster_metrics] Add cluster_metrics data stream for hadoop

### PR DESCRIPTION
**Pull request by yug-elastic**
## What does this PR do?

- Generated the skeleton of Hadoop integration package.
- Added 1 data stream ( Cluster Metrics )
- Added data collection logic.
- Added the ingest pipelines.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added system test cases.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^8.0.0`).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/hadoop directory.
- Run the following command to run tests. 
> elastic-package test

### CODE REVIEW

File: packages/hadoop/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+title: "Hadoop"
+version: 0.1.0
+license: basic
+description: "This Elastic integration collects metrics from hadoop."
+type: integration
+categories:
+  - containers
+  - datastore
+conditions:
+  kibana.version: "^8.0.0"
```
Comment (yug-elastic) [inline packages/hadoop/manifest.yml:12]: This will be updated with 8.2.0 after testing this integration on 8.2.0.

**Comment by yug-elastic** [comment]
This PR is a split of https://github.com/elastic/integrations/pull/2614 as discussed over the comment https://github.com/elastic/integrations/pull/2614#issuecomment-1082704609

File: packages/hadoop/data_stream/cluster_metrics/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        - name: launch_delay_num_ops
+          description: Application Master Launch Delay Operations (Operations)
+          type: long
+        - name: register_delay_avg_time
+          description: Application Master Register Delay Average Time (Milliseconds)
+          type: long
+        - name: register_delay_num_ops
+          description: Application Master Register Delay Operations (Operations)
+          type: long
+    - name: apps
```
Comment (mtojek) [inline packages/hadoop/data_stream/cluster_metrics/fields/fields.yml:20]: nit: applications

File: packages/hadoop/data_stream/cluster_metrics/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
+- name: hadoop.cluster_metrics
```
Comment (mtojek) [inline packages/hadoop/data_stream/cluster_metrics/fields/fields.yml:1]: I suggest to strip the `metrics` suffix and just go with `cluster`.

File: packages/hadoop/data_stream/cluster_metrics/fields/fields.yml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  type: group
+  release: beta
```
Comment (mtojek) [inline packages/hadoop/data_stream/cluster_metrics/fields/fields.yml:3]: `Release tag` is [deprecated now](https://github.com/elastic/package-spec/blob/main/versions/1/manifest.spec.yml#L148). You can drop it from here and from manifests. We will depend on semver prerelease.

File: packages/hadoop/data_stream/cluster_metrics/fields/fields.yml (hunk 3 above)
Comment (yug-elastic) [reply packages/hadoop/data_stream/cluster_metrics/fields/fields.yml:1]: Okay, we will change that.

File: packages/hadoop/data_stream/cluster_metrics/fields/fields.yml (hunk 4 above)
Comment (yug-elastic) [reply packages/hadoop/data_stream/cluster_metrics/fields/fields.yml:3]: Okay, thanks for passing links.

File: packages/hadoop/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+policy_templates:
+  - name: hadoop_metrics
+    title: Hadoop Metrics
+    description: Collect hadoop metrics
+    inputs:
+      - type: http/metrics
+        title: Hadoop Metrics using JMX
+        description: Collect Hadoop Metrics using JMX
+        vars:
+          - name: proxy_url
```
Comment (mtojek) [inline packages/hadoop/manifest.yml:27]: Can we remove the `proxy_url` or will it be removed in other PR?

File: packages/hadoop/data_stream/cluster/fields/fields.yml [hunk 6]
```diff
@@ -1,6 +1,5 @@
-- name: hadoop.cluster_metrics
+- name: hadoop.cluster
   type: group
-  release: beta
   fields:
     - name: application_master
```
Comment (mtojek) [inline packages/hadoop/data_stream/cluster/fields/fields.yml:4]: Didn't we agree on replacing all "master" references with "main"?

File: packages/hadoop/data_stream/cluster/fields/fields.yml [hunk 7]
```diff
@@ -17,7 +16,7 @@
         - name: register_delay_num_ops
           description: Application Master Register Delay Operations (Operations)
```
Comment (mtojek) [inline packages/hadoop/data_stream/cluster/fields/fields.yml:17]: nit: Operations (Operations), a repetition :)

Is the bracket information required or does it add more context?

Comment (yug-elastic) [reply packages/hadoop/data_stream/cluster/fields/fields.yml:17]: It adds the context of unit i.e. number of Operations. Should we discard it?

Comment (mtojek) [reply packages/hadoop/data_stream/cluster/fields/fields.yml:17]: Then I guess you should rephrase it to be more meaningful, as for a person looking at it for the first time it's a mix of words (application, master, delay, operations) without any sense.

File: packages/hadoop/manifest.yml (hunk 5 above)
Comment (masci) [reply packages/hadoop/manifest.yml]: @yug-elastic can you follow up here?

File: packages/hadoop/data_stream/cluster/fields/fields.yml (hunk 7 above)
Comment (masci) [reply packages/hadoop/data_stream/cluster/fields/fields.yml]: @yug-elastic you can say `Register Delay Operations (measured in number of Operations)

Comment (yug-elastic) [reply packages/hadoop/data_stream/cluster/fields/fields.yml]: We have updated it to "... Delay Operations (Number of Operations)". I hope that makes sense.

File: packages/hadoop/manifest.yml (hunk 5 above)
Comment (yug-elastic) [reply packages/hadoop/manifest.yml]: The proxy_url has been removed as per the discussion over https://github.com/elastic/integrations/pull/2955#discussion_r839538174.

**Review by mtojek — CHANGES_REQUESTED** [review]
@yug-elastic Could you please rebase this PR?

**Review by mtojek — APPROVED** [review]

File: packages/hadoop/manifest.yml (hunk 1 above)
Comment (yug-elastic) [reply packages/hadoop/manifest.yml]: Kibana version has been updated to 8.1.0 as discussed over weekly sync for all the integrations.

File: packages/hadoop/data_stream/cluster/manifest.yml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        default: 60s
+      - name: tags
+        type: text
+        title: Tags
+        multi: true
+        required: true
+        show_user: false
+        default:
+          - forwarded
+      - name: ssl
```
Comment (lalit-satapathy) [inline packages/hadoop/data_stream/cluster/manifest.yml:33]: Is the ssl configuration placeholder now or working/used currently?

Comment (yug-elastic) [reply packages/hadoop/data_stream/cluster/manifest.yml:33]: It's not a placeholder and is being used currently.

**Review by lalit-satapathy — APPROVED** [review]
